state/core/src/test/java/com/expediagroup/streamplatform/streamregistry/state/EntityViewUpdaterTest.java

```diff
@@ -18,68 +18,194 @@
   private final DefaultStatus oldStatus = new DefaultStatus();
   private final Entity<DomainKey, DefaultSpecification> oldEntity = entity
-      .withSpecification(oldSpecification)
-      .withStatus(oldStatus);
+    .withSpecification(oldSpecification)
+    .withStatus(oldStatus);
+
+  @After
+  public void clean() {
+    entities.clear();
+  }
 
   @Test
   public void newSpecification() {
     val result = underTest.update(specificationEvent);
 
     assertThat(result, is(nullValue()));
-    assertThat(entities.get(key), is(entity.withStatus(oldStatus)));
+    assertThat(entities.get(key), is(existing(entity.withStatus(oldStatus))));
   }
 
   @Test
   public void updateSpecification() {
-    entities.put(key, oldEntity);
+    entities.put(key, existing(oldEntity));
     val result = underTest.update(specificationEvent);
 
     assertThat(result, is(oldEntity));
-    assertThat(entities.get(key), is(entity.withStatus(oldStatus)));
+    assertThat(entities.get(key), is(existing(entity.withStatus(oldStatus))));
+  }
+
+  @Test
+  public void updateDeletedSpecification() {
+    underTest.update(specificationEvent);
+    underTest.update(specificationDeletionEvent);
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, deleted(entity.withStatus(oldStatus))));
+
+    DefaultSpecification updatedSpecification = specification.withDescription(randomUUID().toString());
+    underTest.update(Event.specification(key, updatedSpecification));
+
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, existing(oldEntity.withSpecification(updatedSpecification))));
   }
 
   @Test
   public void newStatusExistingEntity() {
-    entities.put(key, oldEntity);
+    entities.put(key, existing(oldEntity));
     val result = underTest.update(statusEvent);
 
     assertThat(result, is(oldEntity));
-    assertThat(entities.get(key), is(oldEntity.withStatus(status)));
+    assertThat(entities.get(key), is(existing(oldEntity.withStatus(status))));
   }
 
   @Test
   public void statusNoEntity() {
     val result = underTest.update(statusEvent);
 
     assertThat(result, is(nullValue()));
-    assertThat(entities.get(key), is(nullValue()));
+    assertThat(entities, is(aMapWithSize(0)));
+  }
+
+  @Test
+  public void updateStatusOfDeletedEvent() {
+    underTest.update(specificationDeletionEvent);
+    assertThat(entities, hasEntry(key, deleted(null)));
+
+    val result = underTest.update(statusDeletionEvent);
+    assertThat(result, is(nullValue()));
+    assertThat(entities, hasEntry(key, deleted(null)));
+  }
+
+  @Test
+  public void deleteSpecificationEvent() {
+    val previousEntity = underTest.update(specificationEvent);
+
+    assertThat(previousEntity, is(nullValue()));
+    assertThat(entities.get(key), is(existing(entity.withStatus(oldStatus))));
+    assertThat(entities, hasEntry(key, existing(entity.withStatus(oldStatus))));
+    assertThat(entities, is(aMapWithSize(1)));
```

Comment:
This assertion should probably go first on line 131. It's not worth asserting the rest if this fails